app/helpers/textual_mixins/textual_devices.rb

```diff
@@ -0,0 +1,115 @@
+module TextualMixins::TextualDevices
+  Device = Struct.new(:name, :description, :units, :icon) do
```

Comment:
Please, put the `Device` class into a separate file (as are other `Struct`-based classes).

Not only is it considered a best practice (separate file), but it also prevents all kinds of crazy class-loader issues when reloading files in dev mode.